src/simulators/qasm/qasm_controller.hpp

```diff
@@ -137,6 +137,12 @@ class QasmController : public Base::Controller {
     tensor_network
   };
 
+  // Simulation precision
+  enum class Precision {
+    double_precision,
```

Comment:
This could just be `enum class Precision {single, double}` since it will be used a `Precision::double` / `Precision::single`

Reply:
`double` is reserved in C++.

Reply:
Ah true... could use `Double`/`Single`, but it's not that important